pypdf/generic/_base.py

```diff
@@ -619,6 +619,15 @@ def write_to_stream(
         stream.write(binascii.hexlify(self))
         stream.write(b">")
 
+    def __str__(self) -> str or None:
+        for enc in NameObject.CHARSETS:
+            try:
+                ret = self.decode(enc)
+                return NameObject(ret)
+            except Exception:
```

Comment:
Which actual exceptions can occur here? `Exception` is rather broad and I would have expected to see only `UnicodeDecodeError` here.